src/Paket.Core/Nuget.fs

```diff
@@ -30,6 +33,40 @@ let inline normalizeUrl(url:string) = url.Replace("https","http").Replace("www."
 let cacheFile nugetURL (packageName:PackageName) (version:SemVerInfo) =
     let h = nugetURL |> normalizeUrl |> hash |> abs
     let packageUrl = sprintf "%O.%s.s%d.json" packageName (version.Normalize()) h
-    let cacheFile = FileInfo(Path.Combine(CacheFolder,packageUrl))
-    let errorFile = FileInfo(cacheFile.FullName + ".failed")
-    cacheFile,  errorFile
\ No newline at end of file
+    FileInfo(Path.Combine(CacheFolder,packageUrl))
+
+
+let getDetailsFromCacheOr force nugetURL (packageName:PackageName) (version:SemVerInfo) (get : unit -> NugetPackageCache Async) : NugetPackageCache Async = 
+    let cacheFile = cacheFile nugetURL packageName version
+    let get() = 
+        async {
+            let! result = get()
+            File.WriteAllText(cacheFile.FullName,JsonConvert.SerializeObject(result))
+            return result
+        }
+    async {
+        if not force && cacheFile.Exists then
+            let json = File.ReadAllText(cacheFile.FullName)
+            let result =
+                try
+                    let cachedObject = JsonConvert.DeserializeObject<NugetPackageCache> json
+                    Choice1Of2 cachedObject
```

Comment:
just for sake of keeping the project uniform, can you please use the chessie types? Success and Fail IIRC